### PR TITLE
[clinical-result] Added a11yLabel attribute to clinical-result component

### PR DESCRIPTION
**Pull request by razvanluca**
### Summary

**What was changed:**
Added a11yLabel, role, and focusable flags for clinical-result component

**Why it was changed:**
Needed for accessibility  changes and for Safari and Edge browser support

### Testing

This change was tested using:

- [X] WDIO
- [X] Jest
- [X] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-8882 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-clinical-result/src/ClinicalResultBloodPressure.jsx [hunk 1]
```diff
@@ -149,9 +155,9 @@ const ClinicalResultBloodPressure = (props) => {
     </>
   );
 
-  const modifiedIconElement = hasModifiedIcon && !hasUnverifiedIcon ? (<IconModified className={cx('icon-modified')} />) : null;
-  const commentIconElement = hasCommentIcon && !hasUnverifiedIcon ? (<IconComment className={cx('icon-comment')} />) : null;
-  const unverifiedIconElement = hasUnverifiedIcon ? (<IconUnverified className={cx('icon-unverified')} />) : null;
+  const modifiedIconElement = hasModifiedIcon && !hasUnverifiedIcon ? (<IconModified className={cx('icon-modified')} a11yLabel={intl.formatMessage({ id: 'Terra.clinicalResult.resultModified' })} />) : null;
+  const commentIconElement = hasCommentIcon && !hasUnverifiedIcon ? (<IconComment className={cx('icon-comment')} a11yLabel={intl.formatMessage({ id: 'Terra.clinicalResult.resultComment' })} />) : null;
```
Comment (supreethmr) [inline packages/terra-clinical-result/src/ClinicalResultBloodPressure.jsx:159]:
```suggestion
  const commentIconElement = hasCommentIcon && !hasUnverifiedIcon ? (<IconComment className={cx('icon-comment')} a11yLabel={intl.formatMessage({ id: 'Terra.clinicalResult.resultComment' })} role="img" focusable />) : null;
```
I found that Voice-over needs role to be defined for icons in-order to focus SVG's on safari browser.  [`role="img"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/img_role) would be close match for icon. we can get input from Accessibility team on role !! 

note : `focusable` prop is required for Edge browser in order to make svg focusable

**Review by sdadn — CHANGES_REQUESTED** [review]
Looks like the jests tests are failing due to snapshot errors, which were fixed in #842 . Please rebase with main to fix the build.

File: package.json [hunk 2]
```diff
@@ -73,7 +73,8 @@
     "wdio-lowlight": "terra wdio --themes clinical-lowlight-theme",
     "wdio-fusion": "terra wdio --themes orion-fusion-theme",
     "wdio": "terra wdio --themes terra-default-theme clinical-lowlight-theme orion-fusion-theme",
-    "wdio:docker": "terra wdio --disableSeleniumService=true --themes terra-default-theme clinical-lowlight-theme orion-fusion-theme"
+    "wdio:docker": "WDIO_EXTERNAL_HOST='10.189.160.191' terra wdio --disableSeleniumService=true --themes terra-default-theme clinical-lowlight-theme orion-fusion-theme --browsers chrome --gridUrl grid314.test.devcernerpowerchart.com",
+    "wdio:flowsheet": "WDIO_EXTERNAL_HOST='10.189.160.191' terra wdio --disableSeleniumService=true --themes terra-default-theme clinical-lowlight-theme orion-fusion-theme --spec /packages/terra-clinical-result/tests/wdio/flowsheet-result-cell-spec.js --browsers chrome --gridUrl grid314.test.devcernerpowerchart.com"
```
Comment (sdadn) [inline package.json:77]: What is the reason behind these changes?

**Review by sdadn — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (razvanluca) [reply package.json:77]: that's a mistake... it was for local testing only.

File: package.json [hunk 3]
```diff
@@ -123,4 +123,4 @@
     "webpack-dev-server": "^4.11.1",
     "webpack-merge": "^5.1.4"
   }
-}
+}
```
Comment (sdadn) [inline package.json:126]: nit: newline error

**Review by supreethmr — APPROVED** [review]
+1 with all comments addressed

**Review by sdadn — APPROVED** [review]

**Review by VamsiMure — APPROVED** [review]

**Review by CristinaERoman — COMMENTED** [review]
looks good
